app/Community/Components/UserProfileMeta.php

```diff
@@ -425,19 +426,30 @@ private function calculateAverageFinishedGames(array $userJoinedGamesAndAwards):
         return number_format($averageFinishedGames, 2, '.', '');
     }
 
-    private function calculateAveragePointsPerWeek(string $userMemberSince, int $points = 0): int
+    private function calculateAveragePointsPerWeek(User $user, bool $hardcore, int $points = 0): int
     {
-        $memberSince = Carbon::createFromFormat('Y-m-d H:i:s', $userMemberSince);
+        $field = $hardcore ? "unlocked_hardcore_at" : "unlocked_at";
+
+        $startingDate = $user
+            ->playerAchievements()
+            ->whereNotNull($field)
+            ->orderBy($field)
+            ->first()[$field] ?? null;
+
+        if (is_null($startingDate)) {
+            return 0;
+        }
+
         $now = Carbon::now();
 
-        $weeksOfMembership = $memberSince->diffInWeeks($now);
+        $weeksSinceFirstUnlock = $startingDate->diffInWeeks($now);
```

Comment:
This is unrelated to your change, but we might as well clean this up while we're in here.

I think I may have played with this algorithm a bit when designing the component, and at one point during the design phase there were multiple references to `$now`. However, I believe `$now` is currently an unnecessary abstraction, given it only aliases to `Carbon::now()` and it's only used in one place.

```php
$weeksSinceFirstUnlock = $startingDate->diffInWeeks(Carbon::now());
```